geoips/create_plugin_registries.py

```diff
@@ -14,10 +14,128 @@
+        )
+        for pkg_idx, pkg in enumerate(plugin_packages):
+            if pkg_idx <= comp_idx:  # same package or already compared
+                continue
+            # Track sets of plugins by plugin type
+            # (schemas, yaml_based, and module_based)
+            pkg_registry = yaml.safe_load(
+                open(resources.files(pkg.value) / "registered_plugins.yaml")
+            )
+            for plugin_type in ["yaml_based", "module_based"]:
+                for interface in comp_registry[plugin_type]:
+                    if interface in pkg_registry[plugin_type]:
+                        geoips_plugins = comp_registry[plugin_type][interface]
+                        for plugin in geoips_plugins:
+                            if plugin in pkg_registry[plugin_type][interface]:
+                                pkg_plugin = pkg_registry[plugin_type][interface][
+                                    plugin
+                                ]
+                                if (
+                                    plugin_type == "module_based"
+                                    or "abspath" in pkg_plugin
+                                ):
+                                    remove_registries(plugin_packages)
+                                    raise PluginRegistryError(
+                                        """Error with packages [{}, {}]:
+                                        You can't have two Plugins of the same interface
+                                        [{}] with the same name [{}].""".format(
+                                            comp_pkg.value,
+                                            pkg.value,
+                                            interface,
+                                            plugin,
+                                        )
+                                    )
+                                comp_plugin = comp_registry[plugin_type][interface][
+                                    plugin
+                                ]
+                                for sub_plg in comp_plugin:  # product plugin
+                                    if sub_plg in pkg_plugin:
+                                        remove_registries(plugin_packages)
+                                        raise PluginRegistryError(
+                                            """Error with packages [{}, {}]:
+                                            You can't have two products of the same
+                                            interface [{}] with the same name [{}] found
+                                            under source name [{}].""".format(
+                                                comp_pkg.value,
+                                                pkg.value,
+                                                interface,
+                                                sub_plg,
+                                                plugin,
```

Comment:
Could you include copy/paste of some of the output when you have duplicate plugin names in the body of the PR please?  Ie, with the same name in 2 separate repos, same name both in the geoips repo, same name both in another repo maybe?

Reply:
Sure I can do that for you!

Reply:
This looks good!  Lists both package names if there are duplicates found across 2 different plugin packages, and the single package name if there are duplicate plugins in the same plugin package!

Reply:
This looks good for now. I expect that this will get moved into the PluginRegistry class in #389.